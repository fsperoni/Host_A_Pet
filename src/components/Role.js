import React, { useContext, useState } from "react";
import HostAPetApi from "./Api";
import RoleEditForm from "./RoleEditForm";

const Role = ({ role, roles, setRoles }) => {
  const [showEditForm, setShowEditForm] = useState(false);

  const handleEditRole = () => {
    setShowEditForm(!showEditForm);
  }

  const handleDeleteRole = async(evt) => {
    const id = parseInt(evt.target.children[0].value);
    await HostAPetApi.deleteRole(id);
    setRoles(roles => roles.filter(role => role.id !== id));
  }

  return (
  <div key={role.id} className="card mt-3">
      <div className="card-body">
        <h3>{role.name}</h3>
        <div className="container mt-3">
          <button className="btn btn-sm btn-primary" onClick={handleEditRole}>
            Edit
          </button>
          <button className="btn btn-sm btn-danger ms-2" onClick={handleDeleteRole}>
            <data value={role.id} />
            Delete
          </button>
          {showEditForm ? 
            <RoleEditForm role={role} roles={roles} setRoles={setRoles} setShowEditForm={setShowEditForm}/> : 
          null}
        </div>
      </div>
    </div>
  )
}

export default Role;