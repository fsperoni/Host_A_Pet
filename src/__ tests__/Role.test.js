import { render } from "@testing-library/react";
import Role from "../components/Role";

const {
  roleForm, roles
} = require("./_testCommon");

// smoke test
it('renders without crashing', function () {
  render(
        <Role setRoles={function () { }} role={roleForm()} roles={roles()} />
  );
});

//snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(
        <Role setRoles={function () { }} role={roleForm()} roles={roles()} />
  );
  expect(asFragment()).toMatchSnapshot();
});

it("should have 'Edit' and 'Delete' in the component.", function () {
  const { getByText } = render(
        <Role setRoles={function () { }} role={roleForm()} roles={roles()} />
  );
  expect(getByText("Delete")).toBeInTheDocument();
  expect(getByText("Edit")).toBeInTheDocument();
});
