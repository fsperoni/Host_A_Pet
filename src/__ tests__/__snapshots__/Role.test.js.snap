// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card mt-3"
  >
    <div
      class="card-body"
    >
      <h3>
        Pet Owner
      </h3>
      <div
        class="container mt-3"
      >
        <button
          class="btn btn-sm btn-primary"
        >
          Edit
        </button>
        <button
          class="btn btn-sm btn-danger ms-2"
        >
          <data
            value="2"
          />
          Delete
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
